File: Python3/bulls_and_cows.py
class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        bull = 0
        cows = 0
        
        for i, j in zip(secret, guess):
            if i == j:
                bull += 1
                secret = secret.replace(i, "", 1)
                guess = guess.replace(j, "", 1)
        
        for i, num in enumerate(guess):
            if num in secret:
                cows += 1
                secret = secret.replace(num, "", 1)
        
        return str(bull) + "A" + str(cows) + "B"

File: C++/bulls_and_cows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
        int bull = 0;
        int cows = 0;
        
        for(int i = 0; i < secret.size(); i++) {
            if(secret[i] == guess[i]) {
                bull ++;
                secret[i] = ' ';
                guess[i] = ' ';
            }
        }
        
        for(int i = 0; i < guess.size(); i++) {           
            if(guess[i] != ' ' && secret.find(guess[i]) != string::npos) {
                cows ++;
                secret[secret.find(guess[i])] = ' ';
                guess[i] = ' ';
            }
        }
        
        return to_string(bull) + "A" + to_string(cows) + "B";
    }
};
